### PR TITLE
HCAP-1354 remove Post-hire MOH withdrawal

### PR DESCRIPTION
**Pull request by arranfw**
https://freshworks.atlassian.net/browse/HCAP-1354

- Disabled interest field for hired participants
- Disable withdrawing in service method

### CODE REVIEW

**Comment by taraepp** [comment]
I thought I saw a couple of potential issues so I pulled and ran it locally to make sure:
- I was unable to archive a hired participant (server says: "error: withdrawParticipant is not a function")
- I got some warnings during build that I didn't like the look of: 
image

**Comment by arranfw** [comment]
> I thought I saw a couple of potential issues so I pulled and ran it locally to make sure:
> 
> * I was unable to archive a hired participant (server says: "error: withdrawParticipant is not a function")
> * I got some warnings during build that I didn't like the look of:
>   image

What is this caused by? What was the issue you saw?

**Comment by taraepp** [comment]
> 



> > I thought I saw a couple of potential issues so I pulled and ran it locally to make sure:
> > 
> > * I was unable to archive a hired participant (server says: "error: withdrawParticipant is not a function")
> > * I got some warnings during build that I didn't like the look of:
> >   image
> 
> What is this caused by? What was the issue you saw?

The warnings were present just when I build (make local-run). I switched to master and confirmed I can build master without the warnings. 
 
I double checked the error just now with a fresh participant and I still got it:
- standard hire flow (prospect to site A with HA, hire to site A)
- from hired tab --> archive
- select employment ended, and fill out rest of form (I tried a few different ways: latest was Terminated/Not begun/Yes), then submit form
- server returns 503 Internal Error, toast is "Failed to add participant status". 
- server logs say error: withdrawParticipant is not a function

File: server/services/participants.js [hunk 1]
```diff
@@ -165,12 +166,18 @@ const updateParticipant = async (participantInfo) => {
       { history: participantInfo.history || [], userUpdatedAt: new Date().toJSON() }
     );
     if (changes.interested === 'withdrawn') {
+      // Prevent withdrawal of hired participants HCAP-1354
+      const participantHiredStatuses = await getParticipantHiredStatuses(participantInfo.id);
```
Comment (taraepp) [inline server/services/participants.js:170]: This is what I thought I saw would prevent hiring archived participants. I'm not so sure anymore (I put in a console log here, it didn't log).

File: server/services/participants.js [hunk 2]
```diff
@@ -121,9 +121,19 @@ const getWithdrawnParticipantsBySite = async (siteID) => {
 };
 
 const getParticipantByID = async (id) => {
```
Comment (taraepp) [inline server/services/participants.js:123]: I'd like either a function name change to indicate that it's returning statuses as well, and/or function comment with what it does

**Review by taraepp — COMMENTED** [review]
Just that getParticipantByID function name/comment needs an update, rest looks good to me.

**Comment by arranfw** [comment]
I've decided to drop the server side check for this feature

- participant.interested + participant statuses are too intertwined to easily separate them
- Adding any dependency from participants.js -> participants-status.js causes cyclic dependency errors

**Review by taraepp — APPROVED** [review]
I think this is our safest option that doesn't involve database migrations and/or major refactors. 👍

**Review by fw-noel — APPROVED** [review]

**Pull request merged**